src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
+import React, { FormEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo.type';
+import { TodoList } from './components/TodoList';
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => {
+  const curUser = usersFromServer.find(user => todo.userId === user.id);
+
+  return {
+    ...todo,
+    user: curUser as Todo['user'],
+  };
+});
+
+export const App: React.FC = () => {
+  const [input, setInput] = useState('');
+  const [select, setSelect] = useState('0');
+  const [errors, setErrors] = useState<{ [key: string]: string }>({});
+  const [todos, setTodos] = useState<Todo[]>(todosWithUser || []);
+
+  const validateForm = () => {
+    const newErrors: { [key: string]: string } = {};
+
+    if (!input.trim()) {
+      newErrors.input = 'Please enter a title';
+    }
+
+    if (select === '0') {
+      newErrors.select = 'Please choose a user';
+    }
+
+    return newErrors;
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+
+    const validationErrors = validateForm();
+
+    if (Object.keys(validationErrors).length > 0) {
+      setErrors(validationErrors);
+
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id));
+
+    const newTodo: Todo = {
+      id: maxId + 1,
+      title: input,
+      completed: false,
+      userId: +select,
+      user: usersFromServer.find(user => user.id === +select)!,
```

Comment:
The use of the non-null assertion operator `!` can lead to runtime errors if no user is found. Consider handling the case where `find` returns `undefined` to ensure type safety and avoid potential errors.